### DISCUSSION TITLE
Config file format change?

### DISCUSSION THREAD

**Discussion by BigTP25** [Azure-Player/azure.datafactory.tools#405] [General]
I have csv config files for each environment for our data factories (Dev, UAT, STG, Prod).  I am following the format type, name, path, value like the documentation states.  I have not change the entries in the config files for months...and all of a sudden I am getting this error **[error]ADFT0010: Wrong path defined in config for object(path): linkedService.Nexus(properties.typeProperties.connectionString)_**

Any ideas if anything changed?  Or what I need to do to get this working again?

![ADF bug pic](https://github.com/user-attachments/assets/213b1c97-3d1c-4108-a002-5f0aaba174dd)

**Comment by BigTP25** [comment]
Any help would be greatly appreciated.  The workaround, for now, is to go into this linked service and manually change the connection string for the server it needs to connect.  It would be ideal to have this changed during deployments via the config file but I am stuck ATM with this error.  Removing these entries in the config file results in successful deployments but, like I said, these entries in the config files haven't changed.

**Reply by NowinskiK** [reply]
Nothing has changed in this area and recently there were no new releases, so everything should work as expected.
I'd investigate the environment/agent and any potential changes to the code repo.
Turn on diagnostic mode for an agent to see more details in the log.

**Reply by BigTP25** [reply]
Doesn't the error indicate a problem with the formatting of the entry in the config file?  Can you confirm that my entry is correct from the picture I attached?

**Reply by NowinskiK** [reply]
No, I can't see a formatting problem. Check these examples - it works as part of module's unit tests: 
[test/BigFactorySample2/deployment/config-c003-variables.csv](https://github.com/Azure-Player/azure.datafactory.tools/blob/517d2f0e1c2a8a3aa9e4205845170bde875aed57/test/BigFactorySample2/deployment/config-c003-variables.csv)

Try to simplify your example and run the minimal code on your local machine.

**Reply by BigTP25** [reply]
I took out the entry to specify the URL for SQL Server and just kept the entry to set the default name for the database and I get the same error.  I am really struggling right now...I don't know what else to check.  The error seems to indicate a problem with the entries in the config file but they seem fine.  The deployments work fine if I set the pipeline task to use the stage option vs stating the location of the config file.  

![image](https://github.com/user-attachments/assets/6e7be0b6-b757-4dab-9720-9b84ac2f2624)
![image](https://github.com/user-attachments/assets/b32b1bc6-227c-48fe-a87d-a581c79afeda)

**Reply by BigTP25** [reply]
@NowinskiK I have figure out the issue and its pretty crazy...

The JSON for the linked service is different between two data factories...
This is the JSON for the linked service where I was having problems deploying
![image](https://github.com/user-attachments/assets/d373cf1b-6548-4acf-9cb8-8527b392f5b4)

This is the JSON for the linked service in another ADF for a database on the same logical server...
![image](https://github.com/user-attachments/assets/2e888403-2344-4c40-85c3-c93a8a100e77)

The linked service in both ADFs is an Azure SQL Database but as you can see the JSON behind them is different in two different ADFs.  This is why is was throwing the error it was because the properties changed in the JSON.  

As usual...thank you for being so responsive!!!

**Reply by NowinskiK** [reply]
But... is it the same object (Linked Service)? Which version do you have in your code repository?
The deployment should take ADF & config files from the same folder (location), so that config entries represent correct ADF's objects.
It doesn't matter what is in the target ADF. The deployment works with ADF files from the repo - that should be your source of truth.

**Reply by BigTP25** [reply]
@NowinskiK It's the same object type, Azure SQL Database.  Two separate ADFs, same linked service type, different JSON structure.  The top one is the "new" JSON structure where as the bottom is "old".  Basically, the connectionString property doesn't exist anymore for Azure SQL Databases and the connection string has been broken out into separate property elements in the "new" JSON.

**Reply by NowinskiK** [reply]
This means someone has changed the definition of `nexus` Linked Service - switched from "legacy" (ConnectionString) to "Recommended" (new) on DEV environment / repo:
![image](https://github.com/user-attachments/assets/1b8b1870-06d3-4381-8a00-473d71c92191)

# For record purposes - differences:
## Legacy (ConnectionString)
```
{
    "name": "LS_AzureSqlDb",
    "properties": {
        "annotations": [],
        "type": "AzureSqlDatabase",
        "typeProperties": {
            "connectionString": "Integrated Security=False;Encrypt=True;Connection Timeout=30;Data Source=sqlplayer.database.windows.net;Initial Catalog=master;User ID=sql",
            "password": {
                "type": "AzureKeyVaultSecret",
                "store": {
                    "referenceName": "LS_AzureKeyVault",
                    "type": "LinkedServiceReference"
                },
                "secretName": "sqlserver-sql-password"
            }
        }
    }
}
```

## Recommended (new)
```
{
    "name": "LS_AzureSqlDb",
    "properties": {
        "annotations": [],
        "type": "AzureSqlDatabase",
        "typeProperties": {
            "server": "sqlplayer.database.windows.net",
            "database": "master",
            "encrypt": "mandatory",
            "trustServerCertificate": false,
            "authenticationType": "SQL",
            "userName": "sql",
            "password": {
                "type": "AzureKeyVaultSecret",
                "store": {
                    "referenceName": "LS_AzureKeyVault",
                    "type": "LinkedServiceReference"
                },
                "secretName": "sqlserver-sql-password"
            }
        }
    }
}
```
